Added Smokey Bear Game

theamazing0 opened the pull request:
Slack username: Samvid
Game has a 25 second time limit in which the player tries to optimize their time to collect the best items to help Smokey prepare for hibernation. Players can retry the game to try to beat their old scores.

acornitum commented:
Hey! Running this error even when I have installed requirements.txt. Do you know how to fix this?

<img width="737" alt="image" src="https://github.com/user-attachments/assets/819b456d-e832-4e82-afbe-66f07590435b" />

theamazing0 commented:
Ah looks like I forgot to update the requirements.txt after my update, Should be fixed with latest commit

DominicTYLau commented:
Hi, I’m encountering some errors in the code. Could you please take a look and help me debug it?
<img width="768" alt="Screenshot 2025-02-06 at 12 46 19" src="https://github.com/user-attachments/assets/f4fba01e-c9db-4a98-a0d2-3cf9fda4a666" />

DominicTYLau commented:
Hi, it’s running now and the art is amazing. Please add some instructions on how to play in the README file.

theamazing0 commented:
Thanks, added in the latest commit!

acornitum commented:
I really like the art if your game! The background is especially nice - the colors go really well together, and I like how the birds are animated.

This is close to getting merged, but there are just a few things I want to see you improve on before I do so:
- The bear is currently not very visible along the background - something artists do when having dark objects against dark background is add white highlights to the outline! That will make it more visible but mostly still keep the same colors on the main subject.
- A start screen would be nice! 
- On the end screen, I'd really appreciate it if you wrote something like "score: [number]" instead of just the number so that I know what it represents! On that similar note, maybe make it a different color like white or something so that it's more visible?
- Also, a count down timer would really nice - I didn't really know what the goal of the game was until I read your comment. Let the players know that there is a timer counting down!

Thanks so much for submitting! Can't wait to play when you've done those changes.

_Note: don't open a new PR, just commit changes to this current thread + send a comment when changes have been made!_

theamazing0 commented:
- Added a light colored highlight to the bear
- Added a new screen for start with night/day variants like the mountain backdrop
- added a score label and fixed the color
- added a countdown timer in the form of a border around the screen that slowly disappears, giving a sense of time without having to take up too much place on the actual gameplay area
- also made some fixes to the removal of items after a round that i didn't notice earlier

acornitum commented:
running into an error; get rid of the hw_accel part?

<img width="753" alt="image" src="https://github.com/user-attachments/assets/2adb8c43-9084-44e7-ac8a-ea6ba22b04d2" />

theamazing0 commented:
Make sure you don't have the upstream version of blinka-displayio-pygamedisplay installed and only Cyril's fork which is specified in the requirements.txt. I can remove it on the published version if you would like, but it should work if the correct dependencies are installed. The change wasn't made now, it was present when you tested earlier so I think it's just a dependency version mismatch. Thanks!

acornitum commented:
Nice! Looks much better.

Going to merge it now - fill out the form here: https://forms.hackclub.com/t/isDdJtrnrvus

Thanks for submitting!
